pkg/platform/runtime/buildplan/buildplan.go

```diff
@@ -12,6 +12,78 @@ import (
 	"github.com/go-openapi/strfmt"
 )
 
+type ArtifactListing struct {
+	build            *model.Build
+	runtimeClosure   artifact.Map
+	buildtimeClosure artifact.Map
+	artifactIDs      []artifact.ArtifactID
+}
+
+func NewArtifactListing(build *model.Build) *ArtifactListing {
+	return &ArtifactListing{build: build}
```

Comment:
Would it be possible to calculate the closures and ID's during construction or that's too early since you don't have this context yet?

Reply:
Yeah, we can do that. In some cases, depending on what methods the caller uses we may calculate both but that shouldn't happen.